packages/datepickers/src/styled/StyledDay.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import styled, { DefaultTheme, ThemeProps } from 'styled-components';
+import { retrieveComponentStyles, getColor, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+
+interface IStyledDayProps extends ThemeProps<DefaultTheme> {
+  isPreviousMonth?: boolean;
+  isToday?: boolean;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  isSmall: boolean;
+}
+
+const retrieveStyledDayColor = ({
+  isDisabled,
+  isSelected,
+  isToday,
+  isPreviousMonth,
+  theme
+}: IStyledDayProps) => {
+  if (isDisabled) {
+    return getColor('neutralHue', 400, theme);
+  }
+
+  if (isSelected && !isDisabled) {
+    return theme.colors.background;
+  }
+
+  if (isToday) {
+    return 'inherit';
+  }
+
+  if (isPreviousMonth) {
+    return getColor('neutralHue', 600, theme);
+  }
+
+  return getColor('primaryHue', 600, theme);
+};
+
+const retrieveBackgroundColor = ({ isSelected, isDisabled, theme }: IStyledDayProps) => {
+  if (isSelected && !isDisabled) {
+    return getColor('primaryHue', 600, theme);
+  }
+
+  return 'inherit';
+};
+
+const COMPONENT_ID = 'datepickers.day';
+
+const StyledDay = styled.div.attrs({
+  'data-garden-id': COMPONENT_ID
+})<IStyledDayProps>`
+  display: flex;
+  position: absolute;
+  align-items: center;
+  justify-content: center;
+  border-radius: 50%;
+  background-color: ${retrieveBackgroundColor};
+  cursor: ${props => (props.isDisabled ? 'inherit' : 'pointer')};
+  width: 100%;
+  height: 100%;
+  line-height: ${props => props.theme.lineHeights.md};
+  color: ${retrieveStyledDayColor};
+
+  font-size: ${props => (props.isSmall ? props.theme.fontSizes.sm : props.theme.fontSizes.md)};
+  font-weight: ${props =>
+    props.isToday && !props.isDisabled ? props.theme.fontWeights.semibold : 'inherit'};
+
+  ${props =>
+    !props.isSelected &&
+    !props.isDisabled &&
+    `
+  :hover {
+    background-color: ${getColor('primaryHue', 600, props.theme, 0.08)};
+    color: ${getColor('primaryHue', 800, props.theme)};
+  }
+
+  :active {
+    background-color: ${getColor('primaryHue', 600, props.theme, 0.2)};
+    color: ${getColor('primaryHue', 800, props.theme)};
+  }
+  `}
```

Comment:
The scattered collection of color values throughout this file is difficult to rationalize. After having taken a variety of approaches with CSS ordering (and failed), I want us to consistently the following approach:

1. Base/core CSS properties (display, position, transition, direction, resets, etc.)
1. `${sizeStyles(props)}` – a function that collects all size-related properties (height, width, font-size, line-height, padding, etc) into one location.
    1. associated size pseudo-classes (in LVHFA order) – however, these should be very rare.
1. Psuedo-classes related to base/core properties (LVHFA order)
1. `${colorStyles(props)}` – a function with ALL color styles (border-color, background-color, color, box-shadow)
    1. with psuedo-class properties related to color (LVHFA order)
    1. and color related to `:disabled`/`[aria-disabled]`
1. Base/core `:disabled`/`[aria-disabled]` propreties.
1. Associated styled-component styles (if needed) – repeat the ordering shown above
1. `retrieveComponentStyles`

Again, after having done this for a long time, I believe this gives us the advantage of consolidating the styles most-impacted over time (size & color) into predictable locations. Also, the macro ordering is indicative of what needs to take CSS precedence.

Take a closer look at existing patterns under:
- https://github.com/zendeskgarden/react-components/tree/next/packages/buttons/src/styled
- https://github.com/zendeskgarden/react-components/tree/next/packages/tags/src/styled
- https://github.com/zendeskgarden/react-components/tree/next/packages/typography/src/styled
- https://github.com/zendeskgarden/react-components/tree/jzempel/styled-forms/packages/forms/src/styled (in progress)

Study and repeat what you see there. Let's talk if you feel like those patterns are not working here.